libraries/c_sdk/aws/defender/src/aws_iot_defender_collector.c

```diff
@@ -405,6 +457,212 @@ static void _serializeTcpConnections( void * param1,
+    #if ( configDefenderReportRuntimeStats == 1 )
+        serializerError = _defenderEncoder.openContainerWithKey( &mcuUptimeMap,
+                                                                 KERNEL_TASK_DETAILS,
+                                                                 &taskDetailsArray,
+                                                                 uxArraySize );
+        assertNoError( serializerError );
+
+        for( i = 0; i < uxArraySize; i++ )
+        {
+            IotSerializerEncoderObject_t taskEntryMap = IOT_SERIALIZER_ENCODER_CONTAINER_INITIALIZER_MAP;
+
+            serializerError = _defenderEncoder.openContainer( &taskDetailsArray,
+                                                              &taskEntryMap,
+                                                              7 ); /* define macro. */
+            assertNoError( serializerError );
+
+            /* task_id, task_name, task_status, task_priority, task_abs_cycles, task_percentage, stack_high_watermark. */
+            serializerError = _defenderEncoder.appendKeyValue( &taskEntryMap,
+                                                               KERNEL_TASK_ID,
+                                                               IotSerializer_ScalarSignedInt( pxTaskStatusArray[ i ].xTaskNumber ) );
+            assertNoError( serializerError );
+
+            serializerError = _defenderEncoder.appendKeyValue( &taskEntryMap,
+                                                               KERNEL_TASK_NAME,
+                                                               IotSerializer_ScalarTextString( pxTaskStatusArray[ i ].pcTaskName ) );
+            assertNoError( serializerError );
+
+            serializerError = _defenderEncoder.appendKeyValue( &taskEntryMap,
+                                                               KERNEL_TASK_STATUS,
+                                                               IotSerializer_ScalarSignedInt( pxTaskStatusArray[ i ].eCurrentState ) );
+            assertNoError( serializerError );
+
+            serializerError = _defenderEncoder.appendKeyValue( &taskEntryMap,
+                                                               KERNEL_TASK_PRIORITY,
+                                                               IotSerializer_ScalarSignedInt( pxTaskStatusArray[ i ].uxCurrentPriority ) );
+            assertNoError( serializerError );
+
+            serializerError = _defenderEncoder.appendKeyValue( &taskEntryMap,
+                                                               KERNEL_TASK_ABS_CYCLES,
+                                                               IotSerializer_ScalarSignedInt( pxTaskStatusArray[ i ].ulRunTimeCounter ) );
+            assertNoError( serializerError );
+
+            /* if idle, save the idle time counter. */
+            if( TASK_NAME_EQUAL == _taskNameFilterIdle( pxTaskStatusArray[ i ].pcTaskName ) )
+            {
+                ulIdleTime = pxTaskStatusArray[ i ].ulRunTimeCounter;
```

Comment:
This "if" branch should always be called because idle task should always returned by uxTaskGetSystemState, right?

Reply:
Yes, there's always an idle task in kernel.